### PR TITLE
skinsearch better fixed

### PR DESCRIPTION
**Pull request by bruhmoment21**
preview: https://streamable.com/n4ponw

### CODE REVIEW

**Comment by sserve-kr** [comment]
Can you make not only 'find by name' but also 'find by weapon name'?

**Comment by Cyk-Fad** [comment]
good job

**Comment by Cyk-Fad** [comment]
codes so light

**Comment by Cyk-Fad** [comment]
> Can you make not only 'find by name' but also 'find by weapon name'?

Do it by yourself, just learn C++.
It's easy.

**Comment by dapaster** [comment]
thank you

**Comment by dapaster** [comment]
is it possible to make filter on **ImGui::Combo**

**Comment by bruhmoment21** [comment]
> is it possible to make filter on **ImGui::Combo**

of course . ImGui::BeginCombo and ImGui::EndCombo

**Comment by dapaster** [comment]
> 
> 
> > is it possible to make filter on **ImGui::Combo**
> 
> of course . ImGui::BeginCombo and ImGui::EndCombo

thanks

**Comment by Cyk-Fad** [comment]
My search system because of can search Chinese 
so,I'll still use mine

File: Osiris/Hacks/SkinChanger.cpp [hunk 1]
```diff
@@ -52,6 +52,10 @@ void SkinChanger::initializeKits() noexcept
     std::sort(skinKits.begin(), skinKits.end());
     std::sort(gloveKits.begin(), gloveKits.end());
 
+    // this makes skins with same names work when selecting them
+    for ( size_t i = 0; i < skinKits.size( ); ++i ) skinKits.at( i ).name += "###" + std::to_string( i );
+    for ( size_t i = 0; i < gloveKits.size( ); ++i ) gloveKits.at( i ).name += "###" + std::to_string( i );
```
Comment (danielkrupinski) [inline Osiris/Hacks/SkinChanger.cpp:57]: Use `ImGui::PushID()/ImGui::PopID()` instead.

File: Osiris/GUI.cpp [hunk 2]
```diff
@@ -1013,10 +1013,18 @@ void GUI::renderSkinChangerWindow(bool contentOnly) noexcept
         selected_entry.stat_trak = (std::max)(selected_entry.stat_trak, -1);
         ImGui::SliderFloat("Wear", &selected_entry.wear, FLT_MIN, 1.f, "%.10f", 5);
 
-        ImGui::Combo("Paint Kit", &selected_entry.paint_kit_vector_index, [](void* data, int idx, const char** out_text) {
-            *out_text = (itemIndex == 1 ? SkinChanger::gloveKits : SkinChanger::skinKits)[idx].name.c_str();
-            return true;
-            }, nullptr, (itemIndex == 1 ? SkinChanger::gloveKits : SkinChanger::skinKits).size(), 10);
+        static ImGuiTextFilter filter;
+        filter.Draw( );
```
Comment (danielkrupinski) [inline Osiris/GUI.cpp:1017]: No need to use `ImGuiTextFilter`, filtering using `std::string::find` should work also.

**Comment by danielkrupinski** [comment]
Unfortunately, you deleted the branch, so it's impossible to update the code of this PR.

**Pull request closed without merging**